src/compass/utils/range_split_spectrum.py

```diff
@@ -0,0 +1,118 @@
+import isce3
+import numpy as np
+from isce3.splitspectrum import splitspectrum
+from osgeo import gdal
+
+
+def range_split_spectrum(burst, cfg_split_spectrum,
+                         scratch_path):
+    '''
+    Split burst range spectrum
+    Parameters
+    ----------
+    burst: Sentinel1BurstSlc
+        S1-A/B burst object
+    cfg_split_spectrum: dict
+        Dictionary with split-spetrum options
+    scratch_path: str
+        Directory for storing temp files
+
+    Returns
+    -------
+    burst_raster: isce3.io.Raster
+        3-bands ISCE3 Raster. Band #1: low band;
+        Band #2: main band; Band #3: high band
+    '''
+    length, width = burst.shape
+    lines_per_block = cfg_split_spectrum.lines_per_block
+
+    # In ISCE3, we can use raised cosine to implement S1-A/B Hamming
+    window_type = burst.range_window_type
+    window_type = 'Cosine' if window_type.casefold() == 'hamming' else window_type
+
+    # Extract bandwidths (bw) and create frequency vectors
+    half_bw = 0.5 * burst.range_bandwidth
+    low_bw = cfg_split_spectrum.low_band_bandwidth
+    high_bw = cfg_split_spectrum.high_band_bandwidth
+
+    low_freq_burst = burst.radar_center_frequency - half_bw
+    high_freq_burst = burst.radar_center_frequency + half_bw
+
+    low_band_freqs = np.array([low_freq_burst, low_freq_burst + low_bw])
+    high_band_freqs = np.array([high_freq_burst - high_bw, high_freq_burst])
+    low_center_freq = low_freq_burst + low_bw / 2
+    high_center_freq = high_freq_burst - high_bw / 2
+
+    # Initialize the split-spectrum parameter object. Note, constructor
+    # requires frequency (A/B) but this is not used anywhere below
+    rdr_grid = burst.as_isce3_radargrid()
+    split_spectrum_params = splitspectrum.SplitSpectrum(
```

Comment:
SplitSpectrum is beyond a struct to store some parameters. It is a class that has some methods which does some computations. I suggest to change the name of the instantiated object to better represent the object. My suggestion is:

```suggestion
    split_spectrum_object = splitspectrum.SplitSpectrum(
```